app/Platform/Actions/BuildGameListAction.php

```diff
@@ -57,35 +75,55 @@ public function execute(
             ? $this->getPlayerGames($user, $entries->pluck('id'))
             : collect();
 
-        $backlogGames = $user
-            ? $this->getBacklogGames($user, $entries->pluck('id'))
-            : collect();
+        // If the user is authenticated, pull all their backlog records for the games in the list.
+        // Otherwise, call collect(), which is basically a noop.
+        // We also skip this query if the user is viewing their Want to Play Games List.
+        // In that case, we optimistically assume every game viewed is a "backlog game".
+        $backlogGames = collect();
+        if ($listType !== GameListType::UserPlay && $user) {
+            $backlogGames = $this->getBacklogGames($user, $entries->pluck('id'));
+        }
 
-        $transformedEntries = $entries->getCollection()->map(function (Game $game) use ($playerGames, $backlogGames): GameListEntryData {
-            $playerGame = $playerGames->get($game->id);
-
-            return new GameListEntryData(
-                game: GameData::from($game)->include(
-                    'system.nameShort',
-                    'system.iconUrl',
-                    'achievementsPublished',
-                    'pointsTotal',
-                    'pointsWeighted',
-                    'releasedAt',
-                    'releasedAtGranularity',
-                    'lastUpdated',
-                    'numVisibleLeaderboards',
-                    'numUnresolvedTickets',
-                ),
-                playerGame: $playerGame
-                    ? PlayerGameData::fromPlayerGame($playerGame)->include('highestAward')
-                    : null,
-                isInBacklog: $backlogGames->has($game->id),
-            );
-        });
+        $transformedEntries = $entries
+            ->getCollection()
+            ->map(function (Game $game) use (
+                $listType,
+                $user,
+                $playerGames,
+                $backlogGames
+            ): GameListEntryData {
+                $playerGame = $playerGames->get($game->id);
+
+                return new GameListEntryData(
+                    game: GameData::from($game)->include(
+                        'system.nameShort',
+                        'system.iconUrl',
+                        'achievementsPublished',
+                        'badgeUrl',
+                        'pointsTotal',
+                        'pointsWeighted',
+                        'releasedAt',
```

Comment:
sorting by releaseAt puts "unknown" at the end of the list for both ascending and descending. Not sure if it would be desirable to have them first ascending (as if year was 0)

Reply:
This differs from what I'm seeing on my local. Initiating any new sort should automatically pop the table back to page 1 without needing a page refresh:

![Screenshot 2024-09-23 at 6 49 49 PM](https://github.com/user-attachments/assets/4e83d5b9-2056-49ca-8fa0-79f7d3a201c9)

With that in mind (and assuming you see the same behavior), how do you feel about "unknown" entries ordering at last ascending (as if year was 9999)?

I'm a bit torn on what the right course of action is. I've scoured the web trying to dig up as much UX research on this subject as I can, but [the consensus is mixed](https://ux.stackexchange.com/questions/70265/when-sorting-by-date-do-dateless-entries-come-first-or-last) and seems to always lean towards "it just depends on the context".

With that in mind, the only two contexts I can think of are [Chronological Guide to RetroAchievements](https://retroachievements.org/viewtopic.php?t=25851&o=0) and huge game databases like [IGDB](https://www.igdb.com/platforms/saturn/games). I would assume fans of the chronological guide are going to reach for sorting by earliest first, and seeing a bunch of unknown entries there would probably be perceived as clutter. IGDB also places unknown dates at the end of their ascending list.

I will defer to what you think we should do, but felt like this was a case probably worth making before we ultimately decide.

Reply:
>  Initiating any new sort should automatically pop the table back to page 1 without needing a page refresh:

That was working fine. I was just testing the various sorts with one page of results, and noticed that sorting by release date always left the null values at the bottom.

Games without achievements appear first when sorting ascending by achievements, points, and rarity, so I just assumed null dates would also sort first.

I hadn't considered the player experience for this. As a player, I'd most likely look for _either_ the oldest game on the list, or the newest, and wouldn't care at all about the ones without a date. As such, maybe it's better the way you had it.

Reply:
I see. To be clear, I'm only seeing null dates appear at the bottom of the list if the sort is by earliest (ascending).

If the sort is by latest, they appear at the top:
![Screenshot 2024-09-25 at 5 25 37 PM](https://github.com/user-attachments/assets/5dbd9771-a5f8-4082-bc1d-a73a3be4a3b2)

However, based on the findings reported in my previous comment, this is a behavior I'm in favor of.